### PR TITLE
[SPARK-14245] [Web UI] Display the user in the application view

### PR DESCRIPTION
**Pull request by ajbozarth**
## What changes were proposed in this pull request?

The Spark UI (both active and history) should show the user who ran the application somewhere when you are in the application view. This was added under the Jobs view by total uptime and scheduler mode.
## How was this patch tested?

Manual testing

username

### CODE REVIEW

**Comment by ajbozarth** [comment]
Jenkins, please retest this

**Comment by ajbozarth** [comment]
Jenkins, retest this please

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala [hunk 1]
```diff
@@ -58,6 +58,7 @@ private[spark] class SparkUI private (
 
   val killEnabled = sc.map(_.conf.getBoolean("spark.ui.killEnabled", true)).getOrElse(false)
 
+  val userName = System.getProperty("user.name", "<unknown>")
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/SparkUI.scala]: the user is already set in sc.sparkUser we should use that.

Comment (ajbozarth) [inline core/src/main/scala/org/apache/spark/ui/SparkUI.scala]: Thanks, I hadn't seen that, just pushed the fix

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
LGTM but we should consider the case of using HistoryServer.
When we use HistoryServer, SparkContext is not defined so the user name should be always `<unknown>`.
You can use REST based status API to get the user name even though we use HistoryServer.

Supporting HistoryServer can be separated from this PR.

**Comment by tgravescs** [comment]
Actually this jira was for both history and active so if it doesn't work with history we should fix.  Note that the history file has the user in it already since it displays on the main table in the history page - as "Spark User"

**Comment by sarutak** [comment]
Yeah, we can see the user name in the history page and it's gotten using the REST API.

**Comment by ajbozarth** [comment]
I'll add it the history server later today then, I didn't re-test on the history server after the change so I didn't notice.

**Comment by ajbozarth** [comment]
I just pushed an update. I changed how I get the username so it uses the same method for both active and history. This method actually pulls the information from the same source as both sc.sparkUser and the API do at their root, the system env. Except with this method we make sure we always get the correct user; I noticed a race condition when using sc.sparkUser while finding this solution. After starting up the master when two spark applications are started at once, one by the user who started the master and one who didn't, both would appear to been started by the user who started the master. This would only happened on the first apps started, any apps started after would behave as expected. This race condition doesn't exist with my change.

**Comment by ajbozarth** [comment]
retest this please.

**Comment by tgravescs** [comment]
+1

**Pull request closed without merging**